File: apps/mobile/android/app/src/main/java/com/tchat/screens/VideoScreen.kt
package com.tchat.screens

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material.icons.Icons
import androidx.compose.material.icons.filled.*
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.tchat.designsystem.Colors
import com.tchat.designsystem.Spacing

/**
 * Video conferencing and streaming interface screen
 */
@OptIn(ExperimentalMaterial3Api::class)
@Composable
fun VideoScreen() {
    var selectedTab by remember { mutableStateOf(VideoTab.MEETINGS) }
    var showingCreateMeeting by remember { mutableStateOf(false) }
    var isCameraOn by remember { mutableStateOf(false) }
    var isMicOn by remember { mutableStateOf(true) }

    // Mock meetings
    val upcomingMeetings = listOf(
        Meeting("Team Standup", "09:00 AM", "with John, Sarah, Mike", Icons.Default.VideoCall),
        Meeting("Client Review", "02:00 PM", "with ABC Company", Icons.Default.Group),
        Meeting("Project Planning", "04:30 PM", "with Development Team", Icons.Default.EventNote)
    )

    // Mock recordings
    val recordings = listOf(
        Recording("Q3 Review Meeting", "45 min", "Yesterday", Icons.Default.PlayArrow),
        Recording("Product Demo", "28 min", "2 days ago", Icons.Default.PlayArrow),
        Recording("Training Session", "1h 12min", "1 week ago", Icons.Default.PlayArrow)
    )

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(Colors.background)
    ) {
        // Top app bar
        TopAppBar(
            title = {
                Text(
                    text = "Video",
                    fontSize = 24.sp,
                    fontWeight = FontWeight.Bold,
                    color = Colors.textPrimary
                )
            },
            actions = {
                IconButton(onClick = { showingCreateMeeting = true }) {
                    Icon(
                        imageVector = Icons.Default.Add,
                        contentDescription = "Create meeting",
                        tint = Colors.primary
                    )
                }
            },
            colors = TopAppBarDefaults.topAppBarColors(
                containerColor = Colors.background
            )
        )

        // Tab selector
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .background(Colors.surface)
        ) {
            VideoTab.values().forEach { tab ->
                Column(
                    modifier = Modifier
                        .weight(1f)
                        .clickable { selectedTab = tab }
                        .padding(vertical = Spacing.sm),
                    horizontalAlignment = Alignment.CenterHorizontally
                ) {
                    Icon(
                        imageVector = tab.icon,
                        contentDescription = tab.title,
                        tint = if (selectedTab == tab) Colors.primary else Colors.textSecondary,
                        modifier = Modifier.size(16.dp)
                    )
                    Spacer(modifier = Modifier.height(Spacing.xs))
                    Text(
                        text = tab.title,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Medium,
                        color = if (selectedTab == tab) Colors.primary else Colors.textSecondary
                    )
                }
            }
        }

        // Content based on selected tab
        when (selectedTab) {
            VideoTab.MEETINGS -> MeetingsView(meetings = upcomingMeetings)
            VideoTab.RECORDINGS -> RecordingsView(recordings = recordings)
            VideoTab.LIVE -> LiveView(
                isCameraOn = isCameraOn,
                isMicOn = isMicOn,
                onCameraToggle = { isCameraOn = !isCameraOn },
                onMicToggle = { isMicOn = !isMicOn }
            )
        }
    }
}

// MARK: - Video Tab Enum
enum class VideoTab(
    val title: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector
) {
    MEETINGS("Meetings", Icons.Default.EventNote),
    RECORDINGS("Recordings", Icons.Default.PlayArrow),
    LIVE("Live", Icons.Default.VideoCall)
}

// MARK: - Data Classes
data class Meeting(
    val title: String,
    val time: String,
    val participants: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector
)

data class Recording(
    val title: String,
    val duration: String,
    val date: String,
    val icon: androidx.compose.ui.graphics.vector.ImageVector
)

// MARK: - Meetings View
@Composable
private fun MeetingsView(meetings: List<Meeting>) {
    LazyColumn(
        modifier = Modifier.fillMaxWidth(),
        contentPadding = PaddingValues(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.md)
    ) {
        item {
            // Quick join section
            Card(
                modifier = Modifier.fillMaxWidth(),
                shape = RoundedCornerShape(12.dp),
                colors = CardDefaults.cardColors(
                    containerColor = androidx.compose.ui.graphics.Color.White
                ),
                elevation = CardDefaults.cardElevation(
                    defaultElevation = 4.dp
                )
            ) {
                Column(
                    modifier = Modifier.padding(Spacing.md)
                ) {
                    Text(
                        text = "Quick Join",
                        fontSize = 18.sp,
                        fontWeight = FontWeight.Bold,
                        color = Colors.textPrimary
                    )

                    Spacer(modifier = Modifier.height(Spacing.sm))

                    Row(
                        modifier = Modifier.fillMaxWidth(),
                        verticalAlignment = Alignment.CenterVertically
                    ) {
                        OutlinedTextField(
                            value = "",
                            onValueChange = { },
                            placeholder = {
                                Text(
                                    text = "Enter meeting ID",
                                    color = Colors.textSecondary
                                )
                            },
                            modifier = Modifier.weight(1f),
                            shape = RoundedCornerShape(12.dp)
                        )

                        Spacer(modifier = Modifier.width(Spacing.sm))

                        Button(
                            onClick = { /* Join meeting */ },
                            colors = ButtonDefaults.buttonColors(
                                containerColor = Colors.primary
                            ),
                            shape = RoundedCornerShape(12.dp)
                        ) {
                            Text(
                                text = "Join",
                                color = Colors.textOnPrimary,
                                fontWeight = FontWeight.SemiBold
                            )
                        }
                    }
                }
            }
        }

        item {
            Text(
                text = "Upcoming Meetings",
                fontSize = 18.sp,
                fontWeight = FontWeight.Bold,
                color = Colors.textPrimary
            )
        }

        items(meetings) { meeting ->
            MeetingCard(meeting = meeting)
        }
    }
}

// MARK: - Meeting Card Component
@Composable
private fun MeetingCard(meeting: Meeting) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = androidx.compose.ui.graphics.Color.White
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 4.dp
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.md),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Meeting icon
            Box(
                modifier = Modifier
                    .size(48.dp)
                    .background(
                        color = Colors.primary.copy(alpha = 0.1f),
                        shape = RoundedCornerShape(12.dp)
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = meeting.icon,
                    contentDescription = meeting.title,
                    tint = Colors.primary,
                    modifier = Modifier.size(24.dp)
                )
            }

            Spacer(modifier = Modifier.width(Spacing.md))

            // Meeting info
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = meeting.title,
                    fontSize = 16.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Colors.textPrimary
                )
                Text(
                    text = meeting.time,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    color = Colors.primary
                )
                Text(
                    text = meeting.participants,
                    fontSize = 12.sp,
                    color = Colors.textSecondary
                )
            }

            // Join button
            Button(
                onClick = { /* Join meeting */ },
                colors = ButtonDefaults.buttonColors(
                    containerColor = Colors.primary
                ),
                shape = RoundedCornerShape(8.dp),
                contentPadding = PaddingValues(horizontal = Spacing.md, vertical = Spacing.xs)
            ) {
                Text(
                    text = "Join",
                    fontSize = 12.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = Colors.textOnPrimary
                )
            }
        }
    }
}

// MARK: - Recordings View
@Composable
private fun RecordingsView(recordings: List<Recording>) {
    LazyColumn(
        modifier = Modifier.fillMaxWidth(),
        contentPadding = PaddingValues(Spacing.md),
        verticalArrangement = Arrangement.spacedBy(Spacing.sm)
    ) {
        items(recordings) { recording ->
            RecordingCard(recording = recording)
        }
    }
}

// MARK: - Recording Card Component
@Composable
private fun RecordingCard(recording: Recording) {
    Card(
        modifier = Modifier.fillMaxWidth(),
        shape = RoundedCornerShape(12.dp),
        colors = CardDefaults.cardColors(
            containerColor = androidx.compose.ui.graphics.Color.White
        ),
        elevation = CardDefaults.cardElevation(
            defaultElevation = 2.dp
        )
    ) {
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(Spacing.md),
            verticalAlignment = Alignment.CenterVertically
        ) {
            // Thumbnail placeholder
            Box(
                modifier = Modifier
                    .size(width = 60.dp, height = 45.dp)
                    .background(
                        color = Colors.surface,
                        shape = RoundedCornerShape(8.dp)
                    ),
                contentAlignment = Alignment.Center
            ) {
                Icon(
                    imageVector = recording.icon,
                    contentDescription = "Play",
                    tint = Colors.primary,
                    modifier = Modifier.size(24.dp)
                )
            }

            Spacer(modifier = Modifier.width(Spacing.md))

            // Recording info
            Column(modifier = Modifier.weight(1f)) {
                Text(
                    text = recording.title,
                    fontSize = 14.sp,
                    fontWeight = FontWeight.Medium,
                    color = Colors.textPrimary,
                    maxLines = 1
                )
                Row {
                    Text(
                        text = recording.duration,
                        fontSize = 12.sp,
                        color = Colors.textSecondary
                    )
                    Text(
                        text = " • ",
                        fontSize = 12.sp,
                        color = Colors.textSecondary
                    )
                    Text(
                        text = recording.date,
                        fontSize = 12.sp,
                        color = Colors.textSecondary
                    )
                }
            }

            // More options
            IconButton(onClick = { /* More options */ }) {
                Icon(
                    imageVector = Icons.Default.MoreVert,
                    contentDescription = "More",
                    tint = Colors.textSecondary
                )
            }
        }
    }
}

// MARK: - Live View
@Composable
private fun LiveView(
    isCameraOn: Boolean,
    isMicOn: Boolean,
    onCameraToggle: () -> Unit,
    onMicToggle: () -> Unit
) {
    Column(
        modifier = Modifier.fillMaxSize()
    ) {
        Spacer(modifier = Modifier.weight(1f))

        // Video preview
        Box(
            modifier = Modifier
                .fillMaxWidth()
                .height(300.dp)
                .padding(horizontal = Spacing.md)
                .background(
                    color = Colors.surface,
                    shape = RoundedCornerShape(16.dp)
                ),
            contentAlignment = Alignment.Center
        ) {
            Column(
                horizontalAlignment = Alignment.CenterHorizontally
            ) {
                Icon(
                    imageVector = if (isCameraOn) Icons.Default.Videocam else Icons.Default.VideocamOff,
                    contentDescription = if (isCameraOn) "Camera Active" else "Camera Off",
                    tint = if (isCameraOn) Colors.primary else Colors.textSecondary,
                    modifier = Modifier.size(48.dp)
                )
                Spacer(modifier = Modifier.height(Spacing.sm))
                Text(
                    text = if (isCameraOn) "Camera Active" else "Camera Off",
                    fontSize = 16.sp,
                    fontWeight = FontWeight.Medium,
                    color = if (isCameraOn) Colors.textPrimary else Colors.textSecondary
                )
            }
        }

        Spacer(modifier = Modifier.weight(1f))

        // Control buttons
        Row(
            modifier = Modifier
                .fillMaxWidth()
                .padding(bottom = Spacing.xl),
            horizontalArrangement = Arrangement.SpaceEvenly
        ) {
            // Mic button
            FloatingActionButton(
                onClick = onMicToggle,
                containerColor = if (isMicOn) Colors.success else Colors.surface,
                modifier = Modifier.size(56.dp)
            ) {
                Icon(
                    imageVector = if (isMicOn) Icons.Default.Mic else Icons.Default.MicOff,
                    contentDescription = if (isMicOn) "Mute" else "Unmute",
                    tint = if (isMicOn) Colors.textOnPrimary else Colors.error,
                    modifier = Modifier.size(24.dp)
                )
            }

            // Camera button
            FloatingActionButton(
                onClick = onCameraToggle,
                containerColor = if (isCameraOn) Colors.success else Colors.surface,
                modifier = Modifier.size(56.dp)
            ) {
                Icon(
                    imageVector = if (isCameraOn) Icons.Default.Videocam else Icons.Default.VideocamOff,
                    contentDescription = if (isCameraOn) "Turn off camera" else "Turn on camera",
                    tint = if (isCameraOn) Colors.textOnPrimary else Colors.error,
                    modifier = Modifier.size(24.dp)
                )
            }

            // End call button
            FloatingActionButton(
                onClick = { /* End call */ },
                containerColor = Colors.error,
                modifier = Modifier.size(56.dp)
            ) {
                Icon(
                    imageVector = Icons.Default.CallEnd,
                    contentDescription = "End call",
                    tint = androidx.compose.ui.graphics.Color.White,
                    modifier = Modifier.size(24.dp)
                )
            }
        }
    }
}

// MARK: - Preview
@Preview(showBackground = true)
@Composable
fun VideoScreenPreview() {
    VideoScreen()
}

File: apps/mobile/ios/Sources/Screens/VideoScreen.swift
//
//  VideoScreen.swift
//  TchatApp
//
//  Created by Claude on 22/09/2024.
//

import SwiftUI

/// Video conferencing and streaming interface screen
public struct VideoScreen: View {
    @State private var selectedTab: VideoTab = .meetings
    @State private var showingCreateMeeting = false
    @State private var isCameraOn = false
    @State private var isMicOn = true

    private let colors = Colors()
    private let spacing = Spacing()

    // Mock meetings
    private let upcomingMeetings = [
        ("Team Standup", "09:00 AM", "with John, Sarah, Mike", "video.fill"),
        ("Client Review", "02:00 PM", "with ABC Company", "person.2.fill"),
        ("Project Planning", "04:30 PM", "with Development Team", "calendar"),
    ]

    // Mock recent recordings
    private let recordings = [
        ("Q3 Review Meeting", "45 min", "Yesterday", "play.rectangle.fill"),
        ("Product Demo", "28 min", "2 days ago", "play.rectangle.fill"),
        ("Training Session", "1h 12min", "1 week ago", "play.rectangle.fill"),
    ]

    public init() {}

    public var body: some View {
        NavigationView {
            VStack(spacing: 0) {
                // Segmented control
                HStack(spacing: 0) {
                    ForEach(VideoTab.allCases, id: \.self) { tab in
                        Button(action: {
                            selectedTab = tab
                        }) {
                            VStack(spacing: spacing.xs) {
                                Image(systemName: tab.icon)
                                    .font(.system(size: 16))
                                Text(tab.title)
                                    .font(.system(size: 12, weight: .medium))
                            }
                            .foregroundColor(selectedTab == tab ? colors.primary : colors.textSecondary)
                            .frame(maxWidth: .infinity)
                            .padding(.vertical, spacing.sm)
                        }
                    }
                }
                .background(colors.surface)

                // Content based on selected tab
                switch selectedTab {
                case .meetings:
                    MeetingsView(meetings: upcomingMeetings)
                case .recordings:
                    RecordingsView(recordings: recordings)
                case .live:
                    LiveView(isCameraOn: $isCameraOn, isMicOn: $isMicOn)
                }
            }
            .navigationTitle("Video")
            .navigationBarTitleDisplayMode(.large)
            .toolbar {
                ToolbarItem(placement: .navigationBarTrailing) {
                    Button(action: {
                        showingCreateMeeting = true
                    }) {
                        Image(systemName: "plus.circle.fill")
                            .foregroundColor(colors.primary)
                    }
                }
            }
        }
        .navigationViewStyle(StackNavigationViewStyle())
        .sheet(isPresented: $showingCreateMeeting) {
            CreateMeetingView()
        }
    }
}

// MARK: - Video Tab Enum
enum VideoTab: CaseIterable {
    case meetings, recordings, live

    var title: String {
        switch self {
        case .meetings: return "Meetings"
        case .recordings: return "Recordings"
        case .live: return "Live"
        }
    }

    var icon: String {
        switch self {
        case .meetings: return "calendar"
        case .recordings: return "play.rectangle.fill"
        case .live: return "video.fill"
        }
    }
}

// MARK: - Meetings View
private struct MeetingsView: View {
    let meetings: [(String, String, String, String)]

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        ScrollView {
            VStack(spacing: spacing.md) {
                // Quick join section
                VStack(alignment: .leading, spacing: spacing.sm) {
                    Text("Quick Join")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(colors.textPrimary)
                        .frame(maxWidth: .infinity, alignment: .leading)

                    HStack {
                        TextField("Enter meeting ID", text: .constant(""))
                            .textFieldStyle(PlainTextFieldStyle())
                            .padding(.horizontal, spacing.md)
                            .padding(.vertical, spacing.sm)
                            .background(colors.surface)
                            .cornerRadius(12)

                        Button(action: {}) {
                            Text("Join")
                                .font(.system(size: 14, weight: .semibold))
                                .foregroundColor(colors.textOnPrimary)
                                .padding(.horizontal, spacing.lg)
                                .padding(.vertical, spacing.sm)
                                .background(colors.primary)
                                .cornerRadius(12)
                        }
                    }
                }
                .padding(.horizontal, spacing.md)

                // Upcoming meetings
                VStack(alignment: .leading, spacing: spacing.sm) {
                    Text("Upcoming Meetings")
                        .font(.system(size: 18, weight: .bold))
                        .foregroundColor(colors.textPrimary)
                        .frame(maxWidth: .infinity, alignment: .leading)
                        .padding(.horizontal, spacing.md)

                    ForEach(Array(meetings.enumerated()), id: \.offset) { index, meeting in
                        MeetingCard(
                            title: meeting.0,
                            time: meeting.1,
                            participants: meeting.2,
                            icon: meeting.3
                        )
                    }
                }
            }
            .padding(.top, spacing.sm)
        }
    }
}

// MARK: - Meeting Card Component
private struct MeetingCard: View {
    let title: String
    let time: String
    let participants: String
    let icon: String

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        HStack(spacing: spacing.md) {
            // Meeting icon
            Image(systemName: icon)
                .font(.system(size: 24))
                .foregroundColor(colors.primary)
                .frame(width: 48, height: 48)
                .background(colors.primary.opacity(0.1))
                .cornerRadius(12)

            // Meeting info
            VStack(alignment: .leading, spacing: spacing.xs) {
                Text(title)
                    .font(.system(size: 16, weight: .semibold))
                    .foregroundColor(colors.textPrimary)

                Text(time)
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(colors.primary)

                Text(participants)
                    .font(.system(size: 12))
                    .foregroundColor(colors.textSecondary)
            }

            Spacer()

            // Join button
            Button(action: {}) {
                Text("Join")
                    .font(.system(size: 12, weight: .semibold))
                    .foregroundColor(colors.textOnPrimary)
                    .padding(.horizontal, spacing.md)
                    .padding(.vertical, spacing.xs)
                    .background(colors.primary)
                    .cornerRadius(8)
            }
        }
        .padding(spacing.md)
        .background(Color.white)
        .cornerRadius(12)
        .shadow(color: colors.shadowLight, radius: 4, y: 2)
        .padding(.horizontal, spacing.md)
    }
}

// MARK: - Recordings View
private struct RecordingsView: View {
    let recordings: [(String, String, String, String)]

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        ScrollView {
            VStack(spacing: spacing.sm) {
                ForEach(Array(recordings.enumerated()), id: \.offset) { index, recording in
                    RecordingCard(
                        title: recording.0,
                        duration: recording.1,
                        date: recording.2,
                        icon: recording.3
                    )
                }
            }
            .padding(.horizontal, spacing.md)
            .padding(.top, spacing.sm)
        }
    }
}

// MARK: - Recording Card Component
private struct RecordingCard: View {
    let title: String
    let duration: String
    let date: String
    let icon: String

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        HStack(spacing: spacing.md) {
            // Thumbnail placeholder
            Image(systemName: icon)
                .font(.system(size: 24))
                .foregroundColor(colors.primary)
                .frame(width: 60, height: 45)
                .background(colors.surface)
                .cornerRadius(8)

            // Recording info
            VStack(alignment: .leading, spacing: spacing.xs) {
                Text(title)
                    .font(.system(size: 14, weight: .medium))
                    .foregroundColor(colors.textPrimary)
                    .lineLimit(1)

                HStack {
                    Text(duration)
                        .font(.system(size: 12))
                        .foregroundColor(colors.textSecondary)

                    Text("•")
                        .font(.system(size: 12))
                        .foregroundColor(colors.textSecondary)

                    Text(date)
                        .font(.system(size: 12))
                        .foregroundColor(colors.textSecondary)
                }
            }

            Spacer()

            // More options
            Button(action: {}) {
                Image(systemName: "ellipsis")
                    .foregroundColor(colors.textSecondary)
            }
        }
        .padding(.horizontal, spacing.md)
        .padding(.vertical, spacing.sm)
        .background(Color.white)
        .cornerRadius(12)
        .shadow(color: colors.shadowLight, radius: 2, y: 1)
    }
}

// MARK: - Live View
private struct LiveView: View {
    @Binding var isCameraOn: Bool
    @Binding var isMicOn: Bool

    private let colors = Colors()
    private let spacing = Spacing()

    var body: some View {
        VStack(spacing: spacing.lg) {
            Spacer()

            // Video preview
            RoundedRectangle(cornerRadius: 16)
                .fill(colors.backgroundSecondary)
                .frame(height: 300)
                .overlay(
                    VStack {
                        if isCameraOn {
                            Image(systemName: "video.fill")
                                .font(.system(size: 48))
                                .foregroundColor(colors.primary)
                            Text("Camera Active")
                                .font(.system(size: 16, weight: .medium))
                                .foregroundColor(colors.textPrimary)
                        } else {
                            Image(systemName: "video.slash.fill")
                                .font(.system(size: 48))
                                .foregroundColor(colors.textSecondary)
                            Text("Camera Off")
                                .font(.system(size: 16, weight: .medium))
                                .foregroundColor(colors.textSecondary)
                        }
                    }
                )
                .padding(.horizontal, spacing.md)

            Spacer()

            // Control buttons
            HStack(spacing: spacing.xl) {
                // Mic button
                Button(action: {
                    isMicOn.toggle()
                }) {
                    Image(systemName: isMicOn ? "mic.fill" : "mic.slash.fill")
                        .font(.system(size: 24))
                        .foregroundColor(isMicOn ? colors.textOnPrimary : colors.error)
                        .frame(width: 56, height: 56)
                        .background(isMicOn ? colors.success : colors.surface)
                        .cornerRadius(28)
                }

                // Camera button
                Button(action: {
                    isCameraOn.toggle()
                }) {
                    Image(systemName: isCameraOn ? "video.fill" : "video.slash.fill")
                        .font(.system(size: 24))
                        .foregroundColor(isCameraOn ? colors.textOnPrimary : colors.error)
                        .frame(width: 56, height: 56)
                        .background(isCameraOn ? colors.success : colors.surface)
                        .cornerRadius(28)
                }

                // End call button
                Button(action: {}) {
                    Image(systemName: "phone.down.fill")
                        .font(.system(size: 24))
                        .foregroundColor(.white)
                        .frame(width: 56, height: 56)
                        .background(colors.error)
                        .cornerRadius(28)
                }
            }
            .padding(.bottom, spacing.xl)
        }
        .background(colors.background)
    }
}

// MARK: - Create Meeting View (Placeholder)
private struct CreateMeetingView: View {
    @Environment(\.presentationMode) var presentationMode

    private let colors = Colors()

    var body: some View {
        NavigationView {
            VStack {
                Text("Create New Meeting")
                    .font(.title2)
                    .foregroundColor(colors.textPrimary)

                Spacer()

                Button("Create") {
                    presentationMode.wrappedValue.dismiss()
                }
                .foregroundColor(colors.primary)
            }
            .navigationTitle("New Meeting")
            .navigationBarTitleDisplayMode(.inline)
            .toolbar {
                ToolbarItem(placement: .navigationBarLeading) {
                    Button("Cancel") {
                        presentationMode.wrappedValue.dismiss()
                    }
                }
            }
        }
    }
}

// MARK: - Preview
#if DEBUG
struct VideoScreen_Previews: PreviewProvider {
    static var previews: some View {
        VideoScreen()
            .previewDisplayName("Video Screen")
    }
}
#endif